Release RST 5.1.1

ksterne opened the pull request:
# Release RST 5.1.1

- [x] PATCH
- [ ] MINOR
- [ ] MAJOR

This is the start of the 5.1.1 release pull request which I'm going off the template used for the 5.1 release pull request.  Also need to consult the release guidelines/checklist that we made up a few years ago (and update some of the guidelines as discussed during the workshop).  Here's the key changes for this release:

- New additions

- New/Updated Radar Information
  - Corrected radar names for Siziwang radars (`sze`, `szw`)
  - Added Hokkaido East (`hok`) tdiff value after data analysis
  - Updated Hankasalmi (`han`) info for Borealis installation
  - Updated SANAE (`san`) info for Mabel installation
  - Updated Iceland East, West (`ice`, `icw`) info after electronics upgrades.

- Updates to Current Binaries and Libraries

- Documentation, Bug fixes and Miscellaneous
  - Following discussions around the 2025 SuperDARN Workshop and following Data Analysis Working Group meetings, the author list has been updated to include all known previous authors.




## Deadline: 10 November 2025

Please comment in the Pull Request conversation if you need additional time to test the release.

## Testing

- [ ] Mac OSx
- [x] Linux deb - Debian, Ubuntu, Mint
- [ ] Linux rpm - OpenSuse, Fedora, CentOS

Do any binaries/libraries require special attention during testing?

## Checklist

- [x] Update radar.dat (if required) Check closer to the testing deadline
- [x] Update hardware files (if required) Check closer to the testing deadline
- [x] Update .rst.version
- [x] Update author list in .zenodo.json check again before merging
- [x] Update author list in AUTHORS.md

## Extra Notes
A description of the release will appear soon.

## Approving the release
After testing, please state which operating system you used and what you tested.

If you found no bugs during testing, please approve the PR.

pasha-ponomarenko committed:
Merge pull request #635 from SuperDARN/main

Post RST 5.1 release merge back into develop

ksterne committed:
Updating Siziwang names based on input from PI

aburrell committed:
Merge pull request #638 from SuperDARN/update_Siziwang_name

Updating Siziwang names

ksterne committed:
Adding space for column alignment

ksterne committed:
Correcting radar name

egthomas committed:
Merge pull request #639 from SuperDARN/update_Siziwang_name

Second updates for Siziwang radar name correction

egthomas committed:
Updating HAN hdw.dat and radar.dat files to reflect Borealis upgrade and resumed operations

mtwalach committed:
Merge pull request #643 from SuperDARN/han_update

Updating radar.dat and HAN hdw.dat files after Borealis upgrade

ksterne committed:
Tdiff, number of range gates update

egthomas committed:
Merge pull request #647 from SuperDARN/update_hdw_dat_san

Hdw.dat.san update

egthomas committed:
Updating Iceland East and West hdw.dat files after USRP hardware upgrade

mtwalach committed:
Merge pull request #648 from SuperDARN/iceland_hdw_update

Updating Iceland East and West hdw.dat files after USRP hardware upgrade

ksterne committed:
Updating version number

ksterne committed:
Adding previous authors to Zenodo citation

ksterne commented:
Hello all, I've added two updates to the release branch.  First is simple of updating the version number to 5.1.1.  The second is based on the conversation at the workshop with regards to authorship where I've added all of the names listed in the `AUTHORS.md` file to the Zenodo json citation.  I pulled ORCIDs and affliations where I could.  It might be good to double check I didn't misplace a comma in this json file so that it works correctly before we make the release.

ksterne committed:
Updating position, altitude based on input feedback

egthomas committed:
Fixing hdw.dat column alignment

ksterne commented:
For those not on the e-mail thread, there's been some discussion about a value to use in updating the `hdw.dat.hok` file that arose late on Nov. 9th/early on the 10th.  I'm waiting resolution on this before making an update and then making a release.

ksterne reviewed (commented):
For future reference, does the spacing here matter for parsing the info or was this more for readability? I don't have a problem, but just want to make sure I don't cause a problem in the future.

(Sorry, I meant for this to reference 2ff5a09 but seems as though it didn't link)

ksterne committed:
Updating tdiff value

egthomas commented:
> For future reference, does the spacing here matter for parsing the info or was this more for readability? I don't have a problem, but just want to make sure I don't cause a problem in the future.
> 
> (Sorry, I meant for this to reference [2ff5a09](https://github.com/SuperDARN/rst/commit/2ff5a09c1b884dc341015b34b2d4117a2bf00d7e) but seems as though it didn't link)

Yes, when I updated the geographic radar locations in the hdw.dat files several years ago the column spacing was made consistent across all of them.

egthomas commented:
Also, the Iceland radar hardware updates seem to be missing from both the release notes above and the DAWG report to the PIs.

ksterne commented:
OK, I was just hoping it wasn't a formatting issue for function, but agree that consistency and readability are good to have.

Apologies for forgetting the Iceland info and thanks for pointing it out.  I really thought I had it in there the first time, so not sure what happened.  I've added it to both places.

egthomas commented:
Should the SANAE radar's PI institution be updated from KwaZulu-Natal to SANSA?  I think that change happened a few years ago.

ksterne commented:
Sorry for the delay, I've just e-mailed Judy Stephenson to confirm which institution should be used for the SANAE radar.  I'm hoping to hear back in the next day or two and then get this release out.

ksterne committed:
Adding SANSA institution for SANAE radar

ksterne commented:
I heard back from Judy late last week that both institutions (SANSA and UKZN) should be used for the SANAE radar as the PI has positions at both.  I've updated the radar.dat file to reflect this change.  I think we're good to go now.  I'll leave this for a day unless someone else would like to merge it in. (Maybe do a double check that I updated the Zenodo authors correctly?)

mtwalach commented:
FYI @ksterne I have been trying to install this clean on a Mac OS (working my way through https://github.com/SuperDARN/rst/pull/651/) but got stuck and haven't had time to revisit. Don't want to let that stop you from merging but just thought I'd let you know where I was at.

egthomas commented:
There are also a few additional hardware file changes that we should discuss, so I would request that we hold off on merging this release until after this week's meeting.

ksterne commented:
@egthomas, is there something that you can preview or pass along that we should hold off on making this release?

ksterne committed:
Author list typo fix

ksterne commented:
Going for release now.  I added a note at the end to reflect the change in authorship since this is the first release where all previous authors will be included in the Zenodo citation.
